iroh-net-report/src/reportgen.rs

```diff
@@ -476,23 +476,16 @@ impl Actor {
         // delay by a bit to wait for UDP STUN to finish, to avoid the probe if
         // it's unnecessary.
         if self.last_report.is_none() {
-            // Even if we're doing a non-incremental update, we may want to try our
-            // preferred relay for captive portal detection.
-            let preferred_relay = self
-                .last_report
-                .as_ref()
-                .and_then(|l| l.preferred_relay.clone());
```

Comment:
Because we're in the `self.last_report.is_none()` branch, it follows that `self.last_report.and_then(...)` will also be `None`.

The rest of the code changes all follow from this.